src/index.ts

```diff
@@ -1,18 +1,16 @@
 import { PluginFunction } from 'hapi';
 import { isEqual as ipIsEqual } from 'ip';
+import { getClientIp } from 'request-ip';
 import { unauthorized } from 'boom';
-import castArray = require('lodash.castarray');
-import some = require('lodash.some');
 
 export const register: PluginFunction<{}> = function(server, options, next) {
   server.auth.scheme('ip-whitelist', function(server, whitelisted: string | string[]) {
-    const list = castArray(whitelisted);
+    const list = whitelisted instanceof Array ? whitelisted : [whitelisted];
 
     return {
       authenticate(request, reply) {
-        // in case you are behind a proxy, use Hapi plugin `therealyou`
-        const { remoteAddress } = request.info;
-        if (some(list, (address) => ipIsEqual(remoteAddress, address))) {
+        const remoteAddress = getClientIp(request);
```

Comment:
Have you tested this?

Reply:
Why do you think using `therealyou` plugin is not enough? Keeping this plugin doing less stuff is what I wanted, going for separation of concerns, just accepting the `remoteAddress`.

Reply:
Yes, have tested this it works. Should it be reverted?

Reply:
I am curious why you think is better to address the `proxy` in this plugin instead of doing what was in the comment, using `therealyou` (it should be better documented).

Reply:
request-ip seems to handle a lot more than only X-Forwarded header. I didn't know therealyou was for HapiJS. You are right, better to separate concerns, the best would be to add request-ip to therealyou instead :)

Reply:
Cool! Removing the lodash dependencies is welcome, though. If you want, commit some changes and maybe add a comment of therealyou to the README and we can merge!

Reply:
Removed request-ip and added section to README.md about using behind proxy with therealyou. Should be ready to merge :)